cpp/src/bitmask/null_mask.cu

```diff
@@ -326,20 +327,25 @@ __global__ void copy_offset_bitmask(bitmask_type *__restrict__ destination,
  * @param source_size Number of bits in each mask in @p source
  * @param number_of_mask_words The number of words of type bitmask_type to copy
  */
-__global__ void offset_bitmask_and(bitmask_type *__restrict__ destination,
-                                   bitmask_type const *const *__restrict__ source,
-                                   size_type const *__restrict__ begin_bit,
-                                   size_type num_sources,
-                                   size_type source_size,
-                                   size_type number_of_mask_words)
+template <typename Binop>
+__global__ void offset_bitmask_binop(Binop op,
+                                     bitmask_type *__restrict__ destination,
+                                     bitmask_type const *const *__restrict__ source,
+                                     size_type const *__restrict__ begin_bit,
+                                     size_type num_sources,
+                                     size_type source_size,
+                                     size_type number_of_mask_words)
 {
   for (size_type destination_word_index = threadIdx.x + blockIdx.x * blockDim.x;
        destination_word_index < number_of_mask_words;
        destination_word_index += blockDim.x * gridDim.x) {
-    bitmask_type destination_word = ~bitmask_type{0};  // All bits 1
-    for (size_type i = 0; i < num_sources; i++) {
-      destination_word &= detail::get_mask_offset_word(
-        source[i], destination_word_index, begin_bit[i], begin_bit[i] + source_size);
+    bitmask_type destination_word = detail::get_mask_offset_word(
+      source[0], destination_word_index, begin_bit[0], begin_bit[0] + source_size);
+    for (size_type i = 1; i < num_sources; i++) {
+      destination_word =
+        op(destination_word,
+           detail::get_mask_offset_word(
+             source[i], destination_word_index, begin_bit[i], begin_bit[i] + source_size));
```

Comment:
As an additional optimization, you could also add computing and returning the null count of the resulting mask. That would save launching a kernel some time in the future.

Reply:
Am I missing a way to integrate this in cleanly? The functions that call this generally return a null mask rmm::buffer rather than dealing directly with a resulting column.